File: BruteForce/18111.py
from collections import deque
from functools import cmp_to_key


def bruteforce(h: int):
    global B
    global save
    B_ = B
    able = True
    time = 0

    for i in range(len(map)):
        for j in range(len(map[i])):
            cur_height = map[i][j]
            if (h > cur_height):
                B_ -= (h - cur_height)
                time += (h-cur_height)
            elif (h < cur_height):
                B_ += (cur_height - h)
                time += 2*(cur_height - h)

    if (B_ >= 0):
        # (시간, 높이)를 저장한다
        save.append((time, h))


def compare(t1: tuple, t2: tuple):
    if (t1[0] < t2[0]):
        return -1
    elif (t1[0] > t2[0]):
        return 1
    else:
        if (t1[1] > t2[1]):
            return -1
        else:
            return 1


N, M, B = map(int, input().split())
save = list()
map = [list(map(int, input().split())) for n in range(N)]

min_height = 999
max_height = -1

for i in range(len(map)):
    for j in range(len(map[i])):
        min_height = min(min_height, map[i][j])
        max_height = max(max_height, map[i][j])

for h in range(min_height, max_height + 1):
    bruteforce(h)

save = sorted(save, key=cmp_to_key(compare))

print(save[0][0], save[0][1])

File: BruteForce/18111.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#define ll long long

using namespace std;

ll N = 0, M = 0, B = 0;
ll min_height = 999, max_height = -1;

vector<vector<ll>> map;
vector<pair<ll, ll>> save;

void bruteforce(ll h)
{
    vector<vector<ll>> map_ = map;
    ll B_ = B;
    bool able = true;
    ll time = 0;
    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < M; j++)
        {
            ll cur_height = map[i][j];
            if (h > cur_height)
            {
                // 블록을 이용해서 쌓는다 (2번 연산)
                B_ -= (h - cur_height);
                time += (h - cur_height);
            }
            else if (h < cur_height)
            {
                // 블록을 빼서 저장한다 (1번 연산)
                B_ += (cur_height - h);
                time += 2 * (cur_height - h);
            }
        }
    }

    if (B_ >= 0)
    {
        // (시간, 높이)를 저장한다
        save.push_back(make_pair(time, h));
    }
}

bool compare(const pair<ll, ll> a, const pair<ll, ll> b)
{
    if (a.first != b.first)
    {
        // 오름차순
        return a.first < b.first;
    }
    else
    {
        // 내림차순
        return a.second > b.second;
    }
}

int main()
{
    cin >> N >> M >> B;

    for (ll i = 0; i < N; i++)
    {
        vector<ll> v;
        for (ll j = 0; j < M; j++)
        {
            ll input = 0;
            cin >> input;
            v.push_back(input);
            min_height = min(min_height, input);
            max_height = max(max_height, input);
        }
        map.push_back(v);
    }

    for (ll h = min_height; h <= max_height; h++)
    {
        bruteforce(h);
    }

    sort(save.begin(), save.end(), compare);

    cout << save[0].first << " " << save[0].second;
}